app/src/main/java/de/danoeh/antennapod/adapter/actionbutton/ItemActionButton.java

```diff
@@ -33,18 +33,22 @@ public int getVisibility() {
     }
 
     @NonNull
-    public static ItemActionButton forItem(@NonNull FeedItem item, boolean isInQueue, boolean allowStream) {
+    public static ItemActionButton forItem(@NonNull FeedItem item, boolean allowStream) {
         final FeedMedia media = item.getMedia();
         if (media == null) {
             return new MarkAsPlayedActionButton(item);
         }
 
         final boolean isDownloadingMedia = DownloadRequester.getInstance().isDownloadingFile(media);
+        final boolean isInQueue = item.isTagged(FeedItem.TAG_QUEUE);
         if (FeedItemUtil.isCurrentlyPlaying(media)) {
             return new PauseActionButton(item);
         } else if (item.getFeed().isLocalFeed()) {
             return new PlayLocalActionButton(item);
         } else if (media.isDownloaded()) {
+            if (item.isPlayed()) {
+                return new DeleteActionButton(item);
```

Comment:
Not really happy with this change, to be honest. See also [this forum post](https://forum.antennapod.org/t/add-option-for-download-when-long-pressing-on-episode/991) - users are already confused enough by a manually configured setting that changes the buttons. Having more different actions on screen could be even more confusing.

Reply:
Hmm I really want this, because it saves a few clicks (navigate to download screen).
I guess this could be made optional, but as I said I think this is actually quite smart as it reminds users to remove downloaded episodes so they don't complain about storage piling up.

Reply:
Also I think a long click on the button should open up either the other buttons or a setting for the buttons, as in other parts of the app, so would you prefer a settings dialog on long press where the "smart delete button" can be dis/enabled, and also stream/download default can be changed?

Reply:
Couldn't this be done with auto-delete or with a swipe action?

Reply:
Hmm i guess so, but that's hard to discover, I think I would prefer to do it like this but offer the option to disable it in a long click settings dialog. I think the default is to download>listen>delete, so the default would make that add easy as possible with the option to change it if one has a different workflow (like keep old/favorite episodes). The clear advantage is that it reminds users to delete, this is especially useful with the unified episodes screen where of you quick filter for downloads you can quickly delete already played episodes :)

Reply:
There's a download SwipeAction, maybe it could offer a Snackbar to delete if the episode is already downloaded, the button solution above is still better I think.

Reply:
> this is especially useful with the unified episodes screen where if you quick filter for downloaded

I guess that's your main reason ;) I think the screen should handle this (just like the current downloads screen) instead of introducing inconsistency in all lists.

Reply:
It sure is :)

Reply:
Alternative solution would be to show the _delete if played button_ in the unified episodes only if it's filtered by "downloaded" might be little  bit around the corner, but could you agree to that?